engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterMetalLayer.mm

```diff
@@ -248,20 +248,26 @@ - (BOOL)isKindOfClass:(Class)aClass {
 }
 
 - (void)setDrawableSize:(CGSize)drawableSize {
-  [_availableTextures removeAllObjects];
-  _front = nil;
-  _totalTextures = 0;
-  _drawableSize = drawableSize;
+  @synchronized(self) {
+    [_availableTextures removeAllObjects];
```

Comment:
Can we add a comment near where the ivars _availableTextures, _totalTextures and _front are declared that access to these must be synchornized?